### PR TITLE
Fix units for all physics processes so that they match.

### PR DESCRIPTION
**Pull request by AaronDonahue**
This commit updates the units for all of the field managed variables
in the src/physics directory so that they match.  This is needed for
1. Accuracy, some units were not correct which can lead to confusion.
2. Coupling between processes, the AD will throw an error if a field
   is registered with the field manager with more than one set of units.

There is also some cosmetic changes, like fixing indentation and changing the
name of some variables to match the names we have agreed upon in our
field managed variables group discussion.

[BFB]

### CODE REVIEW

**Comment by AaronDonahue** [comment]
@tcclevenger and @brhillman , I put you two down as reviewers to check and make sure I didn't make any mistakes with units, and my cosmetic changes to variable names.

**Review by tcclevenger — COMMENTED** [review]
@AaronDonahue Looks good to me. I'll let Ben approve so he has a chance to look at it before it merges.

**Comment by AaronDonahue** [comment]
@brhillman , when you get a chance can you check the changes I've made to the RRTMGP interface and stand-alone test code to make sure it all looks good.  Then I think we can merge.

File: components/scream/tests/scream_shoc/CMakeLists.txt
Comment (bartgol) [inline components/scream/tests/scream_shoc/CMakeLists.txt:11]: While we're at it, I think the e3sm convention is to _not_ use an underscore between neX and phys grid specs. That is, `blah_ne4np4` rather than `blah_ne4_np4`.

File: components/scream/src/physics/shoc/atmosphere_macrophysics.cpp
Comment (AaronDonahue) [reply components/scream/src/physics/shoc/atmosphere_macrophysics.cpp:62]: This came from following the units we put down on the Confluence page for fields in the field manager.  Perhaps we should leave it as is for now and if we change the units in Confluence we can submit a quick PR to make the two consistent.

File: components/scream/tests/scream_shoc/CMakeLists.txt
Comment (AaronDonahue) [reply components/scream/tests/scream_shoc/CMakeLists.txt:11]: Ahhh yes, good point.  I think we should make the change now and start the good habit.  I'll make the fix and push the changes so we can merge this PR.

**Comment by AaronDonahue** [comment]
@bartgol , @tcclevenger and @brhillman .  This PR may need to be "re-reviewed" now that I've pushed a new commit.  But it should be ready to merge.

**Review by tcclevenger — APPROVED** [review]

**Pull request merged**
